setup.py

```diff
@@ -41,6 +78,32 @@
 if 'all' in sys.warnoptions:
     log.level = logging.DEBUG
 
+
+class GEOSConfig(object):
+    """Interface to config options from the `geos-config` utility
+    """
+
+    def __init__(self, cmd):
+        self.cmd = cmd
+
+    def get(self, option):
+        try:
+            output = subprocess.check_output([self.cmd, option])
+        except OSError as ex:
+            # e.g., [Errno 2] No such file or directory
+            raise OSError(
+                'Could not find geos-config %r: %s' % (self.cmd, ex))
+        if sys.version_info[0] >= 3:
+            result = output.decode('ascii').strip()
+        else:
+            result = output.strip()
+        log.debug('%s %s: %r', self.cmd, option, result)
+        return result
+
+    def version(self):
+        return tuple(map(int, self.get('--version').strip().split('.')))
```

Comment:
Yes... I'll do this more like you were doing in `get_geos_config()`. Basically, the goal is to avoid importing anything from shapely in setup and just use geos-config or command line options and environment vars.